index.js

```diff
@@ -32,22 +30,31 @@ class KeyState extends q.DesktopApp {
   // eslint-disable-next-line class-methods-use-this
   async run() {
     KeyState.log('Running...');
+
+    this.pollingInterval = this.config.pollingInterval || 1000;
+    this.colors = {
+      on: this.config.colorOn || '#FF0000',
+      off: this.config.colorOff || '#00FF00',
+    };
     const key = this.config.key || this.keys.CapsLock.code;
-    return KeyState.getKeyStatus(key)
-      .then((status) => {
-        const enabled = status === 'True';
-        return new q.Signal({
-          points: [
-            [this.generatePoint(enabled)],
-          ],
-          name: `${key} State`,
-          message: `${key} is ${enabled ? 'enabled' : 'disabled'}`,
-        });
-      }).catch((err) => {
-        // TODO: Introduce an error signal
-        // https://www.daskeyboard.io/applet-development/signals#signal-options
-        KeyState.log(err, 'error');
-      });
+
+    try {
+      var enabled = cmk.getModifierState(key);
```

Comment:
While we're in here, might as well use `const` instead of `var`.

Reply:
I changed it to `var` for exactly 1 reason - I've not written Javascript before, and `const` felt wrong, because it'll be executed every loop. If `const` works because it's instantiated every loop, then I agree.